### PR TITLE
Add integrate_data filter

### LINKED ISSUE #2789: Integrate a variable
### Discussed in https://github.com/pyvista/pyvista/discussions/2761

<div type='discussions-op-text'>

<sup>Originally posted by **MostafaMollaali** June 10, 2022</sup>
Is there a way to integrate a variable in Pyvista that is equivalent to what "_[paraview.simple.IntegrateVariables](https://kitware.github.io/paraview-docs/latest/python/paraview.simple.IntegrateVariables.html)_" does in ParaView?</div>

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Closes #2789 .  Adds ability to integrate data in a dataset including area or volume.

### Details

The vtk class [vtkIntegrateAttributes](https://vtk.org/doc/nightly/html/classvtkIntegrateAttributes.html) includes a `SetDivideAllCellDataByVolume()` method, but I have deliberately chosen not to implement it and in fact have explicitly turned it off.  When enabled, _only_ the integrated cell data is divided by the area or volume.  This could be confusing, and it is straightforward for the user to implement themselves.

The return from this filter is a bit weird at first, an `UnstructuredGrid` with 1 point and 1 cell. I considered returning a dict, e.g. `{"data": 99.1, "Area": 2.45}`. But given that we can have name clashes between point and cell data arrays, we'd have to return two dicts.  So I have chosen to keep the default vtk behavior, and tried to explain this multiple times in the documentation and examples.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
I intended to add to the gallery example one that uses a volume, but I ran out of implementation scheme.  This might be a good 'first issue' that we can add.

File: pyvista/core/filters/data_set.py [hunk 1]
```diff
@@ -5132,3 +5132,44 @@ def reflect(
...
+        --------
+        Integrate data on a sphere mesh.
+
+        >>> import pyvista
+        >>> import numpy as np
+        >>> sphere = pyvista.Sphere(theta_resolution=100, phi_resolution=100)
+        >>> sphere.point_data["data"] = 2 * np.ones(sphere.n_points)
+        >>> integrated = sphere.integrate_data()
+
+         There is only 1 point/cell, so access the only value.
```
Comment (akaszynski) [inline pyvista/core/filters/data_set.py:5161]:
```suggestion
        There is only 1 point/cell, so access the only value.
```

File: pyvista/core/filters/data_set.py [hunk 2]
```diff
@@ -5132,3 +5132,44 @@ def reflect(
             inplace=inplace,
             progress_bar=progress_bar,
         )
+
+    def integrate_data(self, progress_bar=False):
+        """Integrate point and cell data.
+
+        Area or volume also provided in point data.
```
Comment (tkoyama010) [inline pyvista/core/filters/data_set.py:5139]:
```suggestion
        Area or volume is also provided in point data.
```

File: pyvista/core/filters/data_set.py [hunk 3]
```diff
@@ -5132,3 +5132,44 @@ def reflect(
             inplace=inplace,
             progress_bar=progress_bar,
         )
+
+    def integrate_data(self, progress_bar=False):
+        """Integrate point and cell data.
```
Comment (tkoyama010) [inline pyvista/core/filters/data_set.py:5137]: It would be good to explicit the algorithm of the integral method. Can we specify the type of algorithm?

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you for adding this.

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**